torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -585,6 +587,19 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
     return backendTypeToBackend_.at(backendType);
   }
 
+  // Return device types supported by this ProcessGroup.
+  // Note: the return type is `Device` rather than `DeviceType` for the purpose
+  // of easy comparison at Python level. The `Device` will have default index
+  // (-1).
+  std::vector<c10::Device> getDeviceTypes() const {
```

Comment:
is it useful to use a static var to initialize this only once?

Reply:
I was hoping I could. But it would not work when there are multiple PGs.
We will have to use the `_world` structure which is a map using PG as key. Too lazy to write code there. But maybe I should.

Reply:
Done.